### PR TITLE
CollectAsSet aggregator

### PR DESCRIPTION
**Pull request by catoverdrive**

### CODE REVIEW

File: src/test/scala/is/hail/expr/ir/AggregatorsSuite.scala [hunk 1]
```diff
@@ -151,6 +151,34 @@ class AggregatorsSuite {
       Map(Call2(0, 0) -> 3L, Call2(0, 1) -> 1L, Call2(1, 1) -> 1L, (null, 2L)))
   }
 
+  @Test def collectAsSetBoolean() {
+    runAggregator(CollectAsSet(), TBoolean(), FastIndexedSeq(true, false, null, true, false), Set(true, false, null))
```
Comment (maccum) [inline src/test/scala/is/hail/expr/ir/AggregatorsSuite.scala]: can you add a test that doesn't have all three possible values?

**Comment by catoverdrive** [comment]
hmm I'm not sure how we're testing combOps right now. @jigold do you know?

**Comment by jigold** [comment]
The combOps aren't being tested in the AggregatorsSuite, which uses the interpreter. I think we could change lines 389-397 in Interpret.Scala to do a combOp by splitting AggElements and creating a new aggregator for each split and then doing a combOp with a fold at the end. Let me know if this doesn't make sense.

**Comment by catoverdrive** [comment]
seems reasonable. I'll do this as a separate PR.

**Pull request merged**